receiver/k8sclusterreceiver/collection/pods.go

```diff
@@ -183,39 +185,60 @@ func getMetadataForPod(pod *corev1.Pod, mc *metadataStore) map[ResourceID]*Kuber
 
 // collectPodJobProperties checks if pod owner of type Job is cached. Check owners reference
 // on Job to see if it was created by a CronJob. Sync metadata accordingly.
-func collectPodJobProperties(pod *corev1.Pod, JobStore cache.Store) map[string]string {
+func collectPodJobProperties(pod *corev1.Pod, JobStore cache.Store, logger *zap.Logger) map[string]string {
 	jobRef := utils.FindOwnerWithKind(pod.OwnerReferences, k8sKindJob)
 	if jobRef != nil {
-		job, ok, _ := JobStore.GetByKey(utils.GetIDForCache(pod.Namespace, jobRef.Name))
-		if ok {
+		job, exists, err := JobStore.GetByKey(utils.GetIDForCache(pod.Namespace, jobRef.Name))
+		if exists {
 			jobObj := job.(*batchv1.Job)
 			if cronJobRef := utils.FindOwnerWithKind(jobObj.OwnerReferences, k8sKindCronJob); cronJobRef != nil {
 				return getWorkloadProperties(cronJobRef, conventions.AttributeK8sCronJob)
 			}
 			return getWorkloadProperties(jobRef, conventions.AttributeK8sJob)
-
 		}
+		handleErrors(err, jobRef, pod.UID, logger)
 	}
 	return nil
 }
 
 // collectPodReplicaSetProperties checks if pod owner of type ReplicaSet is cached. Check owners reference
 // on ReplicaSet to see if it was created by a Deployment. Sync metadata accordingly.
-func collectPodReplicaSetProperties(pod *corev1.Pod, replicaSetstore cache.Store) map[string]string {
+func collectPodReplicaSetProperties(pod *corev1.Pod, replicaSetstore cache.Store, logger *zap.Logger) map[string]string {
 	rsRef := utils.FindOwnerWithKind(pod.OwnerReferences, k8sKindReplicaSet)
 	if rsRef != nil {
-		replicaSet, ok, _ := replicaSetstore.GetByKey(utils.GetIDForCache(pod.Namespace, rsRef.Name))
-		if ok {
+		replicaSet, exists, err := replicaSetstore.GetByKey(utils.GetIDForCache(pod.Namespace, rsRef.Name))
+		if exists {
 			replicaSetObj := replicaSet.(*appsv1.ReplicaSet)
 			if deployRef := utils.FindOwnerWithKind(replicaSetObj.OwnerReferences, k8sKindDeployment); deployRef != nil {
 				return getWorkloadProperties(deployRef, conventions.AttributeK8sDeployment)
 			}
 			return getWorkloadProperties(rsRef, conventions.AttributeK8sReplicaSet)
 		}
+		handleErrors(err, rsRef, pod.UID, logger)
```

Comment:
let's avoid going too far in DRY :)
looking at `handleErrors(err, rsRef, pod.UID, logger)` I would never expect that this function write warnings under assumption it's called with `exist=false`. 
It'd be much more readable if all the error/warning stuff stays in this function

Reply:
Updated to make this less cryptic. Let me know what you think.